tests/test_validators.py

```diff
@@ -988,3 +988,32 @@ def example_root_validator(cls, values):
     ]
 
     assert root_val_values == [{'a': 123, 'b': 'barbar'}, {'a': 1, 'b': 'snap dragonsnap dragon'}, {'b': 'barbar'}]
+
+
+def test_root_validator_skip_on_failure():
+    called = False
+
+    class ModelA(BaseModel):
+        a: int
+
+        @root_validator
+        def example_root_validator(cls, values):
+            nonlocal called
+            called = True
+
+    with pytest.raises(ValidationError):
+        ModelA(a='a')
+    assert called
+    called = False
```

Comment:
```suggestion
    b_called = False
```
to be more explicit.